src/tmgrad.f90

```diff
@@ -66,30 +56,20 @@ subroutine tmgrad(nat,iat,xyz,g,edum)
    gtemp = get_namespace('gradient.tmp')
    open(newunit=igr,file=gfile)
    open(newunit=jgr,file=gtemp)
-   read(igr,'(a)',end=31)a1
-31 continue
+   read(igr,'(a)')a1
    write(jgr,'(a)')a1
-21 read(igr,'(a)',end=41)a
-   if(index(a,'$').eq.0)then
-      write(jgr,'(a)')a
-      call readl(a,xx,nn)
-      if(index(a,'cycle').ne.0)j=idint(xx(1))
-   else
-   41 write(jgr,'(''  cycle = '',i6,4x,''SCF energy ='',F18.11,3x, &
-      &           ''|dE/dxyz| ='',F10.6)')j+1,edum,gsum     
-      do i=1,nat
-         write(jgr,'(3(F20.14,2x),4x,a2)')xyz(1,i),xyz(2,i),xyz(3,i), &
-         &                                toLcSymbol(iat(i))
-      enddo
-      do i=1,nat
-         write(jgr,'(3D22.13)')g(1,i),g(2,i),g(3,i)
-      enddo
-      a='$end'
-      write(jgr,'(a)')a
-      goto 24
-   endif
-   goto 21
-24 continue
+   read(igr,'(a)')a
+   write(jgr,'(''  cycle = '',i6,4x,''SCF energy ='',F18.11,3x, &
+   &           ''|dE/dxyz| ='',F10.6)')j+1,edum,gsum
+   do i=1,nat
+      write(jgr,'(3(F20.14,2x),4x,a2)')xyz(1,i),xyz(2,i),xyz(3,i), &
+      &                                toLcSymbol(iat(i))
+   enddo
+   do i=1,nat
+      write(jgr,'(3D22.13)')g(1,i),g(2,i),g(3,i)
+   enddo
+   a='$end'
+   write(jgr,'(a)')a
```

Comment:
Similar to above, this refactor is incorrect. The gradient file is read in and a new geometry + gradient inserted before the `$end` group.

Reply:
The gradient file is given as:
```
$grad
  cycle =       1    SCF energy = <real> |dE/dxyz| = <real>
  <x1> <y1> <z1> <sym1>
  <x2> <y2> <z2> <sym2>
  ...
  <gx1> <gy1> <gz1>
  <gx2> <gy2> <gz2>
  ...
  cycle =       2    SCF energy = <real> |dE/dxyz| = <real>
  <x1> <y1> <z1> <sym1>
  <x2> <y2> <z2> <sym2>
  ...
  <gx1> <gy1> <gz1>
  <gx2> <gy2> <gz2>
  ...
$end
```

A new gradient entry is append together with the corresponding coordinates when the `$end` group is encountered.

I have a refactored implementation in [`dftd4`](https://github.com/dftd4/dftd4/blob/9762d50418c0d9859f24005e4e63631b75d85792/source/disp_output.f90#L124-L204) and [`s-dftd3`](https://github.com/awvwgk/simple-dftd3/blob/0fbd084df9b6b140f20362fa832d324ea792e32b/src/dftd3/output.f90#L350-L427).

Reply:
I have looked through the code to find places where tmgrad routine is used, but I did not find any. Can I remove this file?

The output of grep is the following:
```$ grep "tmgrad" * -irn
src/tmgrad.f90:18:subroutine tmgrad(nat,iat,xyz,g,edum)
src/tmgrad.f90:97:end subroutine tmgrad
src/meson.build:147:  'tmgrad.f90',
src/CMakeLists.txt:149:  "${dir}/tmgrad.f90"
```

Reply:
Oh, it is dead code, feel free to remove it than.